src/main/java/com/github/cameltooling/lsp/internal/parser/CamelYamlDSLParser.java

```diff
@@ -220,4 +254,46 @@ private String extractUriFromYamlData(String line) {
 		}
 		return null;
 	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String whenToStop,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		return extractLines(textDocumentItem, position, new String[]{whenToStop}, lines, pos, cleanNewLine);
+	}
+
+	private int extractLines(TextDocumentItem textDocumentItem, Position position, String[] whenToStopArray,
+							 StringBuilder lines, int pos, boolean cleanNewLine) {
+		// If it is a multi-line, cleanNewLine decides if we convert it to one very long single line
+		for (Integer lineNo = position.getLine(); lineNo >=0; lineNo--) {
+			String tempLine = parserFileHelperUtil.getLine(textDocumentItem, lineNo);
+
+			for (String whenToStop : whenToStopArray) {
+				if (tempLine.stripLeading().startsWith(whenToStop)) {
+					// Remove the potential '>' character of multiline
+					if (tempLine.stripLeading().substring(whenToStop.length() + 1).stripLeading().startsWith(">")) {
+						tempLine = tempLine.substring(0, tempLine.indexOf(">"));
+					}
```

Comment:
Adding the other multiline indicators but... the problem I see is that by the time we reach them (going backwards) we already built the StringBuilder with all the content.  Either we do two loops (one to read that indicator to know if we do or don't do the newline thing) or we just assume we are ignoring newlines because this is cameluri and that means we don't care about newlines.